docs/architecture/ADR-004-interest-and-reserves.md

```diff
@@ -113,27 +90,23 @@ func (k Keeper) AccrueAllInterest(ctx sdk.Context) error {
 }
 ```
 
-In the codebase, the function above will be written more efficiently with regards to calculating utilization and interest rates. Error handling and iterator details are also omitted for clarity here.
+In the codebase, the function above will be written more efficiently with regards to calculating utilization and interest rates to avoid the iteration over all user borrow positions per block. Error handling and iterator details are also omitted for clarity here.
 
-### Storing Reserves
+### Protocol Reserve (insurance treasury)
 
-The code snippet above also interacts with reserves. The general design of reserves follows:
+Leverage module will build treasury to secure the protocol and act as a protocol insurance.
```

Comment:
```suggestion
The leverage module will build reserves to secure the protocol and act as insurance. These reserves are used during bad debt repayment, which is discussed in ADR 007.

```

Reply:
I wanted specifically to use word _treasury_ rather than _reserves_

Reply:
Reserves are what they are called - in internal documentation, external API, code, etc.

If changing their name (everywhere), we should do it in a separate PR as this one has gotten off-topic.